linq.go

```diff
@@ -1291,3 +1291,56 @@ func (q Query) MaxFloat64() (min float64, err error) {
 	}
 	return q.values[maxIndex].(float64), nil
 }
+
+// SelectMany returns flattens the resulting sequences into one sequence.
+//
+// Example:
+// 	names, err := From(parents).SelectMany(func (p T, idx int) (T, error) {
+//		return p.(*Parent).Children, nil
+// 	}).Results()
+func (q Query) SelectMany(f func(T, int) (T, error)) (r Query) {
+	return q.SelectManyBy(f, func(p T, c T) (T, error) {
+		return c, nil
+	})
+}
+
+// SelectMany returns flattens the resulting sequences into one sequence.
+//
+// resultSelector takes parent element and child element as inputs
+// and returns a value which will be an element in the resulting query.
+//
+// Example:
+// 	names, err := From(parents).SelectManyBy(func (p T, idx int) (T, error) {
+//		return p.(*Parent).Children, nil
+// 	}, func (p T, c T) (T, error) {
+//              return p.(*Parent).Name + ":" + c.(*Child).Name, nil
+//      }).Results()
+func (q Query) SelectManyBy(f func(T, int) (T, error),
+        resultSelector func(T, T) (T, error)) (r Query) {
+
+	if q.err != nil {
+		r.err = q.err
+		return r
+	}
+	if f == nil || resultSelector == nil {
+		r.err = ErrNilFunc
+		return
+	}
+
+	for i, p := range q.values {
+		val, err := f(p, i)
+		if err != nil {
+			r.err = err
+			return r
+		}
+		for _, c := range From(val).values {
```

Comment:
@mkitawaki hmm I don't exactly remember but `T` is just an alias for `interface{}` so anything automatically can be converted to `T`.

In this case I think you're trying to build a `[]T`, in that case I think you're looking for `takeSliceArg()`.